src/core.cpp

```diff
@@ -460,8 +459,24 @@ void Core::run(double runtodate) {
     // ------------------------------------
     // 6. Run all model dates.
     H_LOG( glog, Logger::NOTICE) << "Running..." << endl;
+    
+    
+    // Let visitors record initial model state
+    // Note that just for this initial output, in calling shouldVisit below, we're telling
+    // visitors that spinup is 'true' (which it's not). This is necessary because the ocean
+    // component hasn't had a chance to initialize its active chemistry yet, and without that
+    // calc_revelle() will throw an error. Kind of hacky; but the only alternative I see is
+    // to create a whole new done_with_spinup() signal to the components--a lot of work.
```

Comment:
It's a bit hacky, yes; allows the visitor to collect model state even though the ocean chemistry has not yet been turned on. That last bit is the issue; it occurs the first time the ocean is `run()`. Now, however, the visitor needs to output things and, because we're no longer in spinup, it tries to collect info on chemistry (and fails). So instead, we tell the visitor, one last time, that we're still in spinup.